queries/highlights.scm

```diff
@@ -1,16 +1,20 @@
-[
-    (comment)
-    (note)
-] @comment
+((comment) @comment)
+((note) @comment.note)
 
-[
-    (date)
-    (interval)
-    (quantity)
-] @number
+((date) @number.date)
+((time) @number.time)
+((interval) @number.interval)
+((quantity) @number.quantity)
 
-((account) @field)
-((commodity) @text.literal)
+((account) @field.account)
+((commodity) @text.literal.commodity)
+
+((check_in) @text.literal.check_in)
+(check_in . (date) @number.date.check_in)
+(check_in . (date) . (time) @number.time.check_in)
+((check_out) @text.literal.check_out)
+(check_out . (date) @number.date.check_out)
+(check_out . (date) . (time) @number.time.check_out)
```

Comment:
What's that `.` syntax for? Does it apply e.g. `@number.time.check_out` to `check_out`, `date` and `time` when they are consecutive?

Reply:
I'm still getting more familiar with tree-sitter and possibly use of `.` here is unnecessary or even incorrect.

The ["Anchors" section](https://tree-sitter.github.io/tree-sitter/using-parsers#anchors) in the "Using Parsers" chapter of the tree-sitter documentation states:
> The anchor operator, ., is used to constrain the ways in which child patterns are matched. It has different behaviors depending on where it’s placed inside a query.<br>
> When . is placed before the first child within a parent pattern, the child will only match when it is the first named node in the parent.

Reply:
Thanks for spoon-feeding me the doc, I couldn't find it quickly :)

So in this case, this more or less does what I described. I guess that's fine, other people might want to see e.g. the time and dates highlighted as such, rather than having one highlight group for all of check_in, date and time, but I don't mind either way. Since you did it first, you get to choose :)